testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -12400,17 +12403,26 @@ fn next_block_and_wait_all(
     }
 
     // wait for followers to catch up
+    let timer = Instant::now();
     loop {
         let finished = follower_blocks_processed
             .iter()
             .zip(followers_current.iter())
-            .map(|(blocks_processed, current)| blocks_processed.load(Ordering::SeqCst) <= *current)
-            .fold(true, |acc, loaded| acc && loaded);
+            .map(|(blocks_processed, start_count)| {
+                blocks_processed.load(Ordering::SeqCst) > *start_count
+            })
+            .all(|b| b);
```

Comment:
I think this could be further simplified to:
```rust
let finished = follower_blocks_processed
    .iter()
    .zip(followers_current.iter())
    .all(|(blocks_processed, current)| blocks_processed.load(Ordering::SeqCst) <= *current);
```